pkg/ssh/connection_native.go

```diff
@@ -180,3 +181,56 @@ func nativeConnectionScp(options ConnectionScpOptions) (*ConnectionScpReport, er
 
 	return &ConnectionScpReport{Response: remotePath}, nil
 }
+
+func nativeConnectionDial(options ConnectionDialOptions) (*ConnectionDialReport, error) {
+	dst, uri, err := Validate(options.User, options.Host, options.Port, options.Identity)
+	if err != nil {
+		return nil, err
+	}
+	_, err = ValidateAndConfigure(uri, options.Identity, options.InsecureIsMachineConnection)
+	if err != nil {
+		return nil, err
+	}
+
+	ssh, err := exec.LookPath("ssh")
+	if err != nil {
+		return nil, fmt.Errorf("no ssh binary found")
+	}
+
+	output := &bytes.Buffer{}
+	errors := &bytes.Buffer{}
+	if strings.Contains(uri.Host, "/run") {
```

Comment:
What is this?

Reply:
parsing weirdness, depending on where this is called, we need to parse out only the ssh host rather than the /run/.... thats needed later.

Reply:
But why do you assume `/run`? The socket can be on any part of the filesystem. It should just split of after the the first slash if you want to trim the path. But doing it here feels wrong, the Host path of an url should never include the path in the first place, that is what the Path field is for.